### ISSUE TITLE
Error starting bcfg2-report-collector with django 1.9

### ISSUE THREAD

**Issue by jsteel44** [Bcfg2/bcfg2#357]
I'm using the latest version from git.

With django 1.9 (I tried 1.9.8 and 1.9.9) I get the following error when trying to run bcfg2-report-collector: 

```
Traceback (most recent call last):
  File "/usr/bin/bcfg2-report-collector", line 32, in <module>
    sys.exit(main())
  File "/usr/bin/bcfg2-report-collector", line 17, in main
    parser.parse()
  File "/usr/lib/python2.7/site-packages/Bcfg2/Options/Parser.py", line 346, in parse
    self._parse_config_options()
  File "/usr/lib/python2.7/site-packages/Bcfg2/Options/Parser.py", line 179, in _parse_config_options
    action(self, self.namespace, value)
  File "/usr/lib/python2.7/site-packages/Bcfg2/Options/Actions.py", line 164, in __call__
    result = self._load_component(values)
  File "/usr/lib/python2.7/site-packages/Bcfg2/Options/Actions.py", line 144, in _load_component
    cls = self._import(mod, name)
  File "/usr/lib/python2.7/site-packages/Bcfg2/Options/Actions.py", line 114, in _import
    return getattr(__import__(module, fromlist=[name]), name)
  File "/usr/lib/python2.7/site-packages/Bcfg2/Reporting/Storage/DjangoORM.py", line 23, in <module>
    from Bcfg2.Reporting.models import *
  File "/usr/lib/python2.7/site-packages/Bcfg2/Reporting/models.py", line 77, in <module>
    class Client(models.Model):
  File "/usr/lib/python2.7/site-packages/django/db/models/base.py", line 94, in __new__
    app_config = apps.get_containing_app_config(module)
  File "/usr/lib/python2.7/site-packages/django/apps/registry.py", line 239, in get_containing_app_config
    self.check_apps_ready()
  File "/usr/lib/python2.7/site-packages/django/apps/registry.py", line 124, in check_apps_ready
    raise AppRegistryNotReady("Apps aren't loaded yet.")
django.core.exceptions.AppRegistryNotReady: Apps aren't loaded yet.
```

It works fine with django 1.7.11, and solj says he is using 1.8.7 and that's OK for him.

**Comment by AlexanderS** [comment]
@jsteel44 Could you please try if this branch is fixing your problem: https://github.com/AlexanderS/bcfg2/tree/fix/django1.9/report-collector (esp. something like this commit: https://github.com/AlexanderS/bcfg2/commit/f500167d2aae88d9b2a68fb792a9cad797a9338b)

**Comment by viiru-** [comment]
This does solve the problem for server startup, but bcfg2-admin still has a similar issue. Here is the traceback (against Django 1.10, but the problem seems to be the same anyway):

Traceback (most recent call last):
  File "/usr/sbin/bcfg2-admin", line 10, in <module>
    sys.exit(CLI().run())
  File "/usr/lib/python2.7/dist-packages/Bcfg2/Server/Admin.py", line 1230, in __init__
    parser.parse()
  File "/usr/lib/python2.7/dist-packages/Bcfg2/Options/Parser.py", line 369, in parse
    getattr(component, "options_parsed_hook")()
  File "/usr/lib/python2.7/dist-packages/Bcfg2/Server/Plugins/Metadata.py", line 45, in load_django_models
    Bcfg2.Server.Plugin.PluginDatabaseModel):
  File "/usr/lib/python2.7/dist-packages/django/db/models/base.py", line 105, in __new__
    app_config = apps.get_containing_app_config(module)
  File "/usr/lib/python2.7/dist-packages/django/apps/registry.py", line 237, in get_containing_app_config
    self.check_apps_ready()
  File "/usr/lib/python2.7/dist-packages/django/apps/registry.py", line 124, in check_apps_ready
    raise AppRegistryNotReady("Apps aren't loaded yet.")
django.core.exceptions.AppRegistryNotReady: Apps aren't loaded yet.

**Comment by AlexanderS** [comment]
Server startup wasn't a problem?! The patches should fix the start of the `bcfg2-report-collector`.

I thought on the server was already everything working. The method I used in the `bcfg2-report-collector` is the same like the Metadata stuff was already using. I will look into it.

**Comment by AlexanderS** [comment]
@viiru- @jsteel44 
Can you try #373? Should be working with django-1.9 now.

**Comment by AlexanderS** [comment]
Should be fixed after merging #373 and #345.
